src/document-info.js

```diff
@@ -56,11 +56,16 @@ export function documentInfoForDoc(nodeOrDoc) {
           canonicalUrl = parseUrl(canonicalTag.href).href;
         }
         const pageViewId = getPageViewId(ampdoc.win);
-        return {url, sourceUrl, canonicalUrl, pageViewId};
+        const res = {sourceUrl, canonicalUrl, pageViewId};
+        Object.defineProperty(res, 'sourceUrl', {
```

Comment:
Let's add a getter method instead (of a getter property).

I think the best solution would be to have our `History` class update this field when the URL changes. But that can wait for another time.